### PR TITLE
Change DropDownButtonWidget separator to an image

### LINKED ISSUE #4123: Make vertical separators (on drop-down menus) skinnable
I think white separators don't look very nice on the UI (they don't match buttons borders).

separator1

To solve this, Mailaender told me to modify this line:
https://github.com/OpenRA/OpenRA/blob/fc49950fd417017d9fafac53726cd831f6dd9af6/OpenRA.Game/Widgets/DropDownButtonWidget.cs#L41

He also kindly adviced me to use system pre-defined colors because they are more convenient than RGB values. Example of possible colors: 
- http://www.dotnetperls.com/color-table
- http://msdn.microsoft.com/fr-fr/library/vstudio/system.drawing.color%28v=vs.110%29.aspx

Here are some choices:
516979colors

But instead of modifying this line, Mailaender suggested with due the idea of drawing this separator like the rest of the UI to make it skinnable (because, this separator is currently shared with other mods).

I think it's a good idea to add this separator into dialog.png
And btw, it's not even necessary to draw a new separator in dialog.png files since we can reuse one of the button borders.

Since this separator is a 1x19 px rectangle, here is what its chrome.yaml declarations could be:

```
# Used for vertical separator on drop-down menus
separator: dialog.png
	separator: 512,0,1,19
```

I've been told that the lines that need to be edited are around line 39:
https://github.com/OpenRA/OpenRA/blob/fc49950fd417017d9fafac53726cd831f6dd9af6/OpenRA.Game/Widgets/DropDownButtonWidget.cs#L39

But I am not a developer so.. that's why I am posting this issue.
EDIT: I will try to work on it but it's a long way... :)

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/1303338-make-vertical-separators-on-drop-down-menus-skinnable?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by penev92**
Closes #4123.

### CODE REVIEW

File: OpenRA.Mods.Common/Widgets/DropDownButtonWidget.cs [hunk 1]
```diff
@@ -41,14 +41,11 @@ public override void Draw()
...
 			var rb = RenderBounds;
-			var color = GetColor();
-			var colorDisabled = GetColorDisabled();
 
 			WidgetUtils.DrawRGBA(image, stateOffset + new float2(rb.Right - rb.Height + 4, rb.Top + (rb.Height - image.Bounds.Height) / 2));
 
-			WidgetUtils.FillRectWithColor(new Rectangle(stateOffset.X + rb.Right - rb.Height,
-				stateOffset.Y + rb.Top + 3, 1, rb.Height - 6),
-				IsDisabled() ? colorDisabled : color);
+			var separator = ChromeProvider.GetImage("separator", "separator");
```
Comment (pchote) [inline OpenRA.Mods.Common/Widgets/DropDownButtonWidget.cs]: How about "dropdown", "separator"?  "separator" by itself could be almost anything.

**Comment by penev92** [comment]
Fixed.

File: mods/cnc/chrome.yaml [hunk 2]
```diff
@@ -512,3 +512,6 @@ order-icons: chrome.png
 	stats: 480,96,16,16
 	stats-disabled: 480,112,16,16
 	stats-active: 480,128,16,16
+
+dropdown: chrome.png
+    separator: 64,0,1,19
```
Comment (x-a-n-a-x) [inline mods/cnc/chrome.yaml]: Please modify this line this way:
separator: 64,1,1,19

File: mods/d2k/chrome.yaml [hunk 3]
```diff
@@ -552,3 +552,5 @@ scrollheader-selected: dialog.png
 	corner-bl: 512,127,1,1
 	corner-br: 639,127,1,1
 
+dropdown: dialog.png
+    separator: 512,0,1,19
```
Comment (x-a-n-a-x) [inline mods/d2k/chrome.yaml]: separator: 512,1,1,19

File: mods/ra/chrome.yaml [hunk 4]
```diff
@@ -963,3 +963,6 @@ mainmenu-border: dialog.png
 	corner-tr: 728,387,40,40
 	corner-bl: 648,467,40,40
 	corner-br: 728,467,40,40
+
+dropdown: dialog.png
+    separator: 512,0,1,19
```
Comment (x-a-n-a-x) [inline mods/ra/chrome.yaml]: separator: 512,1,1,19

File: mods/ts/chrome.yaml [hunk 5]
```diff
@@ -803,3 +803,6 @@ mainmenu-border: dialog.png
 	corner-tr: 728,387,40,40
 	corner-bl: 648,467,40,40
 	corner-br: 728,467,40,40
+
+dropdown: dialog.png
+    separator: 512,0,1,19
```
Comment (x-a-n-a-x) [inline mods/ts/chrome.yaml]: separator: 512,1,1,19

**Comment by penev92** [comment]
Updated.

**Comment by Mailaender** [comment]
Works as promised. :white_check_mark: Also a nice polishing. Now that you re-skinned it, I feel the white was out of place all the time. :+1:

**Comment by x-a-n-a-x** [comment]
:+1: :+1: :+1: :+1: :+1: :+1:

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/a46cec68e8f7ef733975b2461cdd223a3a10bd8d...94cc51ac627b5aaa69a295a9c9089320b3c8e9c4)

**Pull request merged**